components/director/internal/domain/oauth20/service.go

```
@@ -0,0 +1,175 @@
+	reqBody := &clientCredentialsRegistrationBody{
+		GrantTypes: defaultGrantTypes,
+		ClientID:   clientID,
+		Scope:      strings.Join(scopes, " "),
+	}
+
+	buffer := &bytes.Buffer{}
+	err := json.NewEncoder(buffer).Encode(&reqBody)
+	if err != nil {
+		return "", errors.Wrap(err, "while encoding body")
+	}
+
+	resp, closeBody, err := s.doRequest(http.MethodPost, s.clientEndpoint, buffer)
+	if err != nil {
+		return "", err
+	}
+	defer closeBody(resp.Body)
+
+	if resp.StatusCode != http.StatusCreated {
+		return "", fmt.Errorf("invalid HTTP status code: received: %d, expected %d", resp.StatusCode, http.StatusCreated)
+	}
+
+	var registrationResp clientCredentialsRegistrationResponse
+	err = json.NewDecoder(resp.Body).Decode(&registrationResp)
+	if err != nil {
+		return "", errors.Wrap(err, "while decoding response body")
+	}
+
+	return registrationResp.ClientSecret, nil
+}
+
+func (s *service) unregisterClient(clientID string) error {
+	endpoint := fmt.Sprintf("%s/%s", s.clientEndpoint, clientID)
+
+	resp, closeBody, err := s.doRequest(http.MethodDelete, endpoint, nil)
+	if err != nil {
+		return err
+	}
+	defer closeBody(resp.Body)
+
+	if resp.StatusCode != http.StatusNoContent {
+		return fmt.Errorf("invalid HTTP status code: received: %d, expected %d", resp.StatusCode, http.StatusNoContent)
+	}
+
+	return nil
+}
+
+func (s *service) doRequest(method string, endpoint string, body io.Reader) (*http.Response, func(body io.ReadCloser), error) {
+	req, err := http.NewRequest(method, endpoint, body)
+	if err != nil {
+		return nil, nil, errors.Wrap(err, "while creating new request")
+	}
+
+	req.Header.Set("Accept", applicationJSONType)
+	req.Header.Set("Content-Type", applicationJSONType)
+
+	resp, err := s.httpCli.Do(req)
+	if err != nil {
+		return nil, nil, errors.Wrapf(err, "while doing request to %s", s.clientEndpoint)
+	}
+
+	closeBodyFn := func(body io.ReadCloser) {
+		if body == nil {
+			return
+		}
+		_, err = io.Copy(ioutil.Discard, resp.Body)
```

you ignore error from Copy